### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by jojo987N**
###  Vet clinic database: query and update animals table
For this milestone :

- Add a column species of type string to animals table. 
- Modify schema.sql file.
-  Add these insertions to data.sql file.
- Inside a transaction update the animals table by setting the species column to unspecified
- Roll back the change and verify that the species columns went back to the state before the transaction.
- Write queries to answer the questions.
- Take a screenshot of the results of queries.
psql-transaction-19
psql-transaction-18
psql-transaction-17
psql-transaction-16
psql-transaction-15
psql-transaction-14
psql-transaction-13
psql-transaction-12
psql-transaction-11
psql-transaction-10
psql-transaction-9
psql-transaction-8
psql-transaction-7
psql-transaction-6
psql-transaction-5
psql-transaction-4
psql-transaction-3
psql-transaction-2
psql-transaction-1

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -18,6 +18,41 @@ SELECT * from animals WHERE name <> 'Gabumon';
 
 SELECT * from animals WHERE weight_kg >= 10.4 and weight_kg <= 17.3;
 
+Begin;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
Comment (omar-muhamad) [inline queries.sql]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=to%20unspecified.-,Verify%20that%20change%20was%20made.%20Then%20roll%20back%20the%20change%20and%20verify%20that%20the%20species%20columns%20went%20back%20to%20the%20state%20before%20the%20transaction.,-Inside%20a%20transaction), Please verify that change was made and verify that the species columns went back after the rollback.

Hint: 
```sql
   BEGIN; 
   UPDATE animals
   SET species = 'unspecified'; 
   SELECT species from animals; -- verify that change was made
   ROLLBACK;
   SELECT species from animals; -- verify that change was undone
```

File: queries.sql [hunk 2]
```diff
@@ -18,6 +18,41 @@ SELECT * from animals WHERE name <> 'Gabumon';
...
+
+Begin;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+commit;
```
Comment (omar-muhamad) [inline queries.sql]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=species%20already%20set.-,Verify%20that%20changes%20were%20made.,Verify%20that%20changes%20persist%20after%20commit.,-Take%20a%20screenshot), Please verify that changes were made and verify that changes persist after commit.

File: queries.sql [hunk 3]
```diff
@@ -18,6 +18,41 @@ SELECT * from animals WHERE name <> 'Gabumon';
...
+
+Begin;
+DELETE FROM animals;
```
Comment (omar-muhamad) [inline queries.sql:36]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=then%20roll%20back%20the%20transaction), Please rollback the transaction.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @jojo987N,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql (hunk 1 above)
Comment (Munsa1) [reply queries.sql]: - [ ] Kindly implement the suggested change by the previous reviewer to verify that the change was made and verify that the species columns went back after the rollback.

e.g
```
   BEGIN; 
   UPDATE animals
   SET species = 'unspecified'; 
   SELECT species from animals; -- verify that change was made
   ROLLBACK;
   SELECT species from animals; -- verify that change was undone
```

File: queries.sql (hunk 2 above)
Comment (Munsa1) [reply queries.sql]: - [ ] Again here, after setting the species column with values, kindly verify that the changes were made by using a `select` statement.
e.g
`SELECT species from animals;`

File: queries.sql (hunk 3 above)
Comment (Munsa1) [reply queries.sql:36]: - [ ] Again here kindly add the rollback statement so as to undo the changes made to the database as requested by the previous reviewer

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @jojo987N, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions

- [ ] Kindly remove the `Live Demo` section from your readme file as you have not deployed the app and therefore you have no link to share.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jojo987N** [comment]
I updated queries.sql file

**Review by Reem-lab — APPROVED** [review]
# Hi @jojo987N,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  No Linters errors 🔥
- [ ]  Great job working on the functionality💯
- [ ]  Kudos for adding the screenshot🔥
- [ ]  Followed GitFlow ⭐🚀


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**